src/dpq2/conv/time.d

```diff
@@ -68,28 +86,89 @@ if( is( T == TimeStampWithoutTZ ) )
     );
 }
 
+/++
+    Structure to represent PostgreSQL's Timestamp without time zone type.
+    It is assumed to be in Local Time.
++/
 struct TimeStampWithoutTZ
 {
-    DateTime dateTime;
+    DateTime dateTime; /// date and time of TimeStamp
     Duration fracSec; /// fractional seconds
 
     invariant()
     {
-        assert(fracSec >= Duration.zero);
-        assert(fracSec < 1.seconds);
+        import std.conv : to;
+        assert(fracSec >= Duration.zero, fracSec.to!string);
+        assert(fracSec < 1.seconds, fracSec.to!string);
     }
 
+    /// Returns the TimeStampWithoutTZ farthest in the future which is representable by TimeStampWithoutTZ.
     static max()
     {
         return TimeStampWithoutTZ(DateTime.max, long.max.hnsecs);
     }
 
+    /// Returns the TimeStampWithoutTZ farthest in the past which is representable by TimeStampWithoutTZ.
     static min()
     {
         return TimeStampWithoutTZ(DateTime.min, Duration.zero);
     }
+
+    /// Converts the value to SysTime with possibility to set the time zone it's in
+    SysTime toSysTime(immutable TimeZone tz = LocalTime()) const
+    {
+        return SysTime(dateTime, fracSec, tz);
+    }
+
+    /++
+        Creates timestamp without timezone from SysTime.
+        It takes the resulting date and time from the SysTime and drops the time zone information.
+    +/
+    static TimeStampWithoutTZ fromSysTime(in SysTime time)
```

Comment:
Sure, it is the whole point of TimeStampWithoutTZ.
But this conversion comes handy if one really wants to store this type in the DB (I won't do that myself exactly because of what you said :) ).
It takes just the resulting time in whatever TZ it is and drops the TZ information.
And it is exactly what happens in PG if you send there for example `2017-1-15 14:29:17.123456789`

So if `Clock.currTime` returns `2017-1-15T14:29:17.123456789+02` (which is in local time if not specified othervise) than after this conversion it will be simply `2017-1-15 14:29:17.123456789`.
If you call `Clock.currTime(UTC())` it will return for this case `2017-1-15T12:29:17.123456789Z`  and after this conversion it will be `2017-1-15 12:29:17.123456789`

One thing that can be modified is that the conversion will convert the provided SysTime to LocalTime and drops it afterwards, but then what if PG is set to different LocalTime? It will mean totally different time. So I just dropped the information about TZ and described it in the documentation assuming that the caller knows what he's doing and sets the SysTime to TZ in which he wants to store it in the DB.
Other solution could be to add the required parameter to specify in what TZ should resulting `TimeStampWithoutTZ` represent and convert to that before the TZ drop. That way caller will be forced to set it.

But I have no problem removing this method altogether and leave it to the user of the library to make sure it works from his side. :)
So just let me know if remove yes/no ;-)

Reply:
> But this conversion comes handy if one really wants to store this type in the DB

There is special type without TZ called `DateTime` - <i>holds both the date and the time. It is optimized for calendar-based operations and has no concept of time zone.</i>

`DateTime` can be converted to it (but with some accuracy lost, as I understand), but currently method for this isn't implemented.

Or should be used `TimeStampWithoutTZ` directly - it is created exactly for this purpose and declared as public.

> So just let me know if remove yes/no ;-)

What about rewrite it to `fromDateTime`?

Reply:
I'll add `opCast` to DateTime as it is direct conversion.
I'm not sure about adding `fromDateTime` as it is equivalent to `TimeStampWithoutTZ(dateTime)`, but can add too of course.

But as DateTime has no fracSecs there are legitimate uses for conversions to/from SysTime.
What about the proposed required TZ in `fromSysTime`? It is a bit weird considering the SysTime already has this specified, but on the other hand it is a way to make sure that the conversion is made correctly to the time in the desired TZ before the information about TZ is lost.

Reply:
> But as DateTime has no fracSecs there are legitimate uses for conversions to/from SysTime.

If user need it he should use dpq2's `TimeStampWithoutTZ` which exactly describes PG's `timestamp` type.

> What about the proposed required TZ in fromSysTime?

Sorry, no. Let application user take responsibility for a possible error to himself. I.e., we should give him ability to convert `SysTime` to `DateTime` or to `TimeStampWithoutTZ` by himself.

Reply:
Ok, I'll remove conversion from SysTime and let the caller decides if he need it